doc/sr_integration.rst

```diff
@@ -0,0 +1,211 @@
+provide a preconditioning step each iteration and post the state of the simulation
+to the database. ::
+
+    program main
+
+        ! Initialize the model, setup MPI, communications, read input files
+        call initialize_model( temperature, number_of_timesteps )
+
+        main_loop: do i=1,number_of_timesteps
+
+            ! Write the current state of the simulation to a file
+            call write_current_state(temperature)
+
+            ! Call a time integrator to step the temperature field forward
+            call timestep_simulation(temperature)
+
+        enddo
+    end program main
+
+Following the guidelines from above, the first step is to initialize the client
+and create a unique identifier for the given processor. This should be done
+within roughly the same portion of the code where the rest of the model. ::
+
+    ! Import SmartRedis modules
+    use, only smartredis_client : client_type
+
+    ! Declare a new variable called client and a string to create a unique
+    ! name for keys
+    type(client_type) :: smartredis_client
+    character(len=7)  :: key_prefix
+    integer :: mpi_rank, mpi_code, smartredis_code
+
+    ! Note adding use_cluster as an additional runtime argument for SmartRedis
+    call initialize_model(temperature, number_of_timesteps, use_cluster)
+    call smartredis_client%initialize(use_cluster)
+    call MPI_Comm_rank(MPI_COMM_WORLD, mpi_rank, mpi_code)
+    write(key_prefix,'(I6.6,A)') mpi_rank, '_'
+
+    ! Assume all ranks will use the same machine learning model
+    if (mpi_rank==0) call set_model_from_file("example_model_key", "path/to/model.pt", "TORCH", "gpu")
+
+Next, add the calls in the main loop to send the temperature to the database ::
+
+    character(len=10), dimension(1) :: model_input, model_output
+
+    main_loop: do i=1,number_of_timesteps
+
+        ! Write the current state of the simulation to a file
+        call write_current_state(temperature)
+        model_input(1) = key_prefix//"temperature"
+        model_output(1) = key_prefix//"temperature_out"
+        call smartredis_client%put_tensor(model_input(1))
+
+        ! Run the machine learning model
+        return_code = smartredis_client%run_model("example_model_key", model_input, model_output)
+        ! The following line overwrites the prognostic temperature array
+        return_code = smartredis_client%unpack_tensor(model_output(1), temperature)
+
+        ! Call a time integrator to step the temperature field forward
+        call timestep_simulation(temperature)
+
+    enddo
+
+Now when this program runs, every time step the client will be used to the
+temperature array in the database, the database will call a machine learning
+model to do the inference, the simulation will request the inference,
+and finally unpack the array into the existing temperature array. For more
+complex examples, please see some of the integrations in the SmartSim Zoo or
+feel free to contact the team at CrayLabs@hpe.com
```

Comment:
Not sure what this means: "Now when this program runs, at every time step, the client will be used to send the
temperature array to the database, " 
slightly unclear

Reply:
Reworded to:

This model will now use the client every timestep to put a
temperature array in the orchestrator, instruct the orchestrator to call
a machine learning model for prediction/inference, and unpack the resulting
inference into the existing temperature array.